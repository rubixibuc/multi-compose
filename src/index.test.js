import React from 'react';
import multiCompose from './index';
import { shallow, mount } from 'enzyme';
import { withProps } from 'recompose';

describe('multi-compose', () => {
  it('should create two left instances of a shared composed component', () => {
    const MultiComponent = multiCompose(
      withProps({
        sharedProp1: 'sharedProp1'
      }),
      withProps({
        sharedProp2: 'sharedProp2'
      })
    )(
      {
        Instance1: [
          withProps({
            instanceProp1: 'instance1Prop1'
          }),
          withProps({
            instanceProp2: 'instance1Prop2'
          })
        ],
        Instance2: [
          withProps({
            instanceProp1: 'instance2Prop1'
          }),
          withProps({
            instanceProp2: 'instance2Prop2'
          })
        ]
      },
      true
    )(
      ({ sharedProp1, sharedProp2, instanceProp1, instanceProp2 }) =>
        `${sharedProp1} ${sharedProp2} ${instanceProp1} ${instanceProp2}`
    );

    expect(MultiComponent.Instance1).toBeDefined();
    expect(MultiComponent.Instance2).toBeDefined();

    const Component1 = mount(<MultiComponent.Instance1 />);
    const Component2 = mount(<MultiComponent.Instance2 />);

    expect(Component1).toMatchSnapshot();
    expect(Component2).toMatchSnapshot();
  });

  it('should create two right instances of a shared composed component', () => {
    const MultiComponent = multiCompose(
      withProps({
        sharedProp1: 'sharedProp1'
      }),
      withProps({
        sharedProp2: 'sharedProp2'
      })
    )({
      Instance1: [
        withProps({
          instanceProp1: 'instance1Prop1'
        }),
        withProps({
          instanceProp2: 'instance1Prop2'
        })
      ],
      Instance2: [
        withProps({
          instanceProp1: 'instance2Prop1'
        }),
        withProps({
          instanceProp2: 'instance2Prop2'
        })
      ]
    })(
      ({ sharedProp1, sharedProp2, instanceProp1, instanceProp2 }) =>
        `${sharedProp1} ${sharedProp2} ${instanceProp1} ${instanceProp2}`
    );

    expect(MultiComponent.Instance1).toBeDefined();
    expect(MultiComponent.Instance2).toBeDefined();

    const Component1 = mount(<MultiComponent.Instance1 />);
    const Component2 = mount(<MultiComponent.Instance2 />);

    expect(Component1).toMatchSnapshot();
    expect(Component2).toMatchSnapshot();
  });
});
