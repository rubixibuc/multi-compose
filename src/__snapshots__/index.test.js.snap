// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi-compose should create two left instances of a shared composed component 1`] = `
<withProps(withProps(withProps(withProps(Component))))>
  <withProps(withProps(withProps(Component)))
    instanceProp1="instance1Prop1"
  >
    <withProps(withProps(Component))
      instanceProp1="instance1Prop1"
      instanceProp2="instance1Prop2"
    >
      <withProps(Component)
        instanceProp1="instance1Prop1"
        instanceProp2="instance1Prop2"
        sharedProp1="sharedProp1"
      >
        <Component
          instanceProp1="instance1Prop1"
          instanceProp2="instance1Prop2"
          sharedProp1="sharedProp1"
          sharedProp2="sharedProp2"
        >
          sharedProp1 sharedProp2 instance1Prop1 instance1Prop2
        </Component>
      </withProps(Component)>
    </withProps(withProps(Component))>
  </withProps(withProps(withProps(Component)))>
</withProps(withProps(withProps(withProps(Component))))>
`;

exports[`multi-compose should create two left instances of a shared composed component 2`] = `
<withProps(withProps(withProps(withProps(Component))))>
  <withProps(withProps(withProps(Component)))
    instanceProp1="instance2Prop1"
  >
    <withProps(withProps(Component))
      instanceProp1="instance2Prop1"
      instanceProp2="instance2Prop2"
    >
      <withProps(Component)
        instanceProp1="instance2Prop1"
        instanceProp2="instance2Prop2"
        sharedProp1="sharedProp1"
      >
        <Component
          instanceProp1="instance2Prop1"
          instanceProp2="instance2Prop2"
          sharedProp1="sharedProp1"
          sharedProp2="sharedProp2"
        >
          sharedProp1 sharedProp2 instance2Prop1 instance2Prop2
        </Component>
      </withProps(Component)>
    </withProps(withProps(Component))>
  </withProps(withProps(withProps(Component)))>
</withProps(withProps(withProps(withProps(Component))))>
`;

exports[`multi-compose should create two right instances of a shared composed component 1`] = `
<withProps(withProps(withProps(withProps(Component))))>
  <withProps(withProps(withProps(Component)))
    sharedProp1="sharedProp1"
  >
    <withProps(withProps(Component))
      sharedProp1="sharedProp1"
      sharedProp2="sharedProp2"
    >
      <withProps(Component)
        instanceProp1="instance1Prop1"
        sharedProp1="sharedProp1"
        sharedProp2="sharedProp2"
      >
        <Component
          instanceProp1="instance1Prop1"
          instanceProp2="instance1Prop2"
          sharedProp1="sharedProp1"
          sharedProp2="sharedProp2"
        >
          sharedProp1 sharedProp2 instance1Prop1 instance1Prop2
        </Component>
      </withProps(Component)>
    </withProps(withProps(Component))>
  </withProps(withProps(withProps(Component)))>
</withProps(withProps(withProps(withProps(Component))))>
`;

exports[`multi-compose should create two right instances of a shared composed component 2`] = `
<withProps(withProps(withProps(withProps(Component))))>
  <withProps(withProps(withProps(Component)))
    sharedProp1="sharedProp1"
  >
    <withProps(withProps(Component))
      sharedProp1="sharedProp1"
      sharedProp2="sharedProp2"
    >
      <withProps(Component)
        instanceProp1="instance2Prop1"
        sharedProp1="sharedProp1"
        sharedProp2="sharedProp2"
      >
        <Component
          instanceProp1="instance2Prop1"
          instanceProp2="instance2Prop2"
          sharedProp1="sharedProp1"
          sharedProp2="sharedProp2"
        >
          sharedProp1 sharedProp2 instance2Prop1 instance2Prop2
        </Component>
      </withProps(Component)>
    </withProps(withProps(Component))>
  </withProps(withProps(withProps(Component)))>
</withProps(withProps(withProps(withProps(Component))))>
`;
